### DISCUSSION TITLE
eslint complains about 'await' outside an async function, but the code works

### DISCUSSION THREAD

**Discussion by danielniccoli** [sveltejs/kit#12025] [Q&A] [answered]
I have the following code `$lib/db.js`.

```js
import { MikroORM } from "@mikro-orm/postgresql"
import config from "$lib/mikro-orm.config"

export const orm = await MikroORM.init(config)
```

eslint complains about a `Parsing error: Cannot use keyword 'await' outside an async function`

I have not enough experience to understand the implications of that message, or if that is just a false-positive, because the code seems to work, despite the error.

What should I do?

**Comment by teemingc** [comment]
The error you’re seeing may be due to how eslint is configured.

I’m not familiar with eslint configs but I do know that top level await (await outside an async function) is only supported for modern environments https://caniuse.com/?search=top%20level%20await it’s worth reviewing your eslint config or browserlist config to see if ecma2022 or newer is set anywhere otherwise I think it will set rules for older environments?

**Reply by danielniccoli** [reply]
I'm using the original `.eslintrc.cjs` from `npm create svelte@latest`. If the top-level await works with that template, should that not also be reflected in the eslint configuration then? From that perspective, I would say that's a bug. What do you think? Should I report that - and where?

**Reply by danielniccoli** [reply]
Btw, that is the original content:

```cjs
/** @type { import("eslint").Linter.Config } */
module.exports = {
    root: true,
    extends: ["eslint:recommended", "plugin:svelte/recommended", "prettier"],
    parserOptions: {
        sourceType: "module",
        ecmaVersion: 2020,
        extraFileExtensions: [".svelte"],
    },
    env: {
        browser: true,
        es2017: true,
        node: true,
    },
}
```

**Reply by teemingc** [reply, marked as answer]
I’m assuming the default eslint config is set to maintain comptibility with older browsers / client devices. If you intend to use more modern javascript features or only on the server, you can simply change the config to allow it or apply an override to files that are suffixed with .server.js, etc. https://eslint.org/docs/latest/use/configure/configuration-files#how-do-overrides-work

**Reply by danielniccoli** [reply]
Makes sense. And the override worked. Thanks!
